contracts/BaseRouter.sol

```diff
@@ -481,6 +481,7 @@ abstract contract BaseRouter is Initializable, IDepositWithReferral {
         address to,
         uint256 minSharesOut
     ) internal returns (uint256 sharesOut) {
+        if (amount == type(uint256).max) amount = IERC20(savingsRate.asset()).balanceOf(address(this));
```

Comment:
Actually better to make this call in the `mixer` function as you already have the address of the token to deposit